### PR TITLE
[evaluation] support UniEval and add CHRF metric

### PR DESCRIPTION
**Pull request by chengeharrison**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

Support UniEval and add CHRF metric.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: applications/Chat/evaluate/README.md [hunk 1]
```diff
@@ -12,12 +12,13 @@ pip install -r requirements.txt
...
 ## Evaluation Pipeline
 
-The whole evaluation pipeline consists of two methods:
+The whole evaluation pipeline consists of three methods:
 
 1. `GPT Evaluation`: evaluates model predictions using GPT models.
    * Compare the performance of two different models (battle).
    * Rate the model according to pre-defined metrics using prompting design.
 2. `Automatic Evaluation`: evaluates model predictions using automatic metrics.
+3. `UniEval`: evaluates model predictions using UniEval models.
```
Comment (TongLi3701) [inline applications/Chat/evaluate/README.md:21]: Please add (English only) for this method.

File: applications/Chat/evaluate/README.md [hunk 2]
```diff
@@ -94,6 +97,25 @@ There are 5 types of automatic evaluation metrics listed in the table below:
 |              Distinct               | Measure the diversity of generation text by counting the unique n-grams. |
 |              BERTScore              | Measure the semantic similarity between tokens of predictions and references with BERT. |
 | Precision<br/> Recall<br/> F1 Score | Measure the number of overlaps between prediction and reference (design for classification and extraction categories). |
+|                CHRF                 | Measure the similarity of character n-grams between prediction and reference. |
+
+### UniEval Evaluation
```
Comment (TongLi3701) [inline applications/Chat/evaluate/README.md:102]: Should have #### here.

File: applications/Chat/evaluate/config/config_cn.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "language": "cn",
+  "path_for_UniEval": {
+    "summarization": "path to unieval-sum",
+    "dialogue": "path to unieval-dialog",
+    "data2text": "path to unieval-sum"
+  },
```
Comment (TongLi3701) [inline applications/Chat/evaluate/config/config_cn.json:7]: As far as I understand, UniEval used T5 as base model and trained only on English corpus. It does not support Chinese.

Comment (TongLi3701) [reply applications/Chat/evaluate/config/config_cn.json:7]: Could you please have a try and see if it works on Chinese text?

Comment (chengeharrison) [reply applications/Chat/evaluate/config/config_cn.json:7]: Yes, UniEval doesn't support Chinese. If we use a translated version of the dialogue [example](https://github.com/maszhongming/UniEval#get-scores-for-dialogue) UniEval gives, the scores are not reasonable. Some scores such as naturalness and coherence shouldn't be so low.

```python
# a list of dialogue histories
src_list = ['嗨，你知道很多互联网的事情吗？ \n 我知道很多不同的网站和一些网站设计，你呢？ \n\n']
# a list of additional context that should be included into the generated response
context_list = ['应用程序和网站上的3条横线菜单被称为汉堡包按钮。\n']
# a list of model outputs to be evaluated
output_list = ['我也是。你知道应用程序和网站上的3条横线菜单被称为汉堡包按钮吗？']
```
Output:
```
+-------------------+----------+
|     Dimensions    |  Score   |
+-------------------+----------+
|    naturalness    | 0.205913 |
|     coherence     | 0.25297  |
|    engagingness   | 0.981161 |
|    groundedness   | 0.987428 |
| understandability | 0.196486 |
|      overall      | 0.524792 |
+-------------------+----------+
```

For task summarization [example](https://github.com/maszhongming/UniEval#get-scores-for-summarization), the scores are also not reasonable. Some scores such as fluency and relevance shouldn't be so high.

```python
# a list of source documents
src_list = ['彼得和伊丽莎白在晚上乘坐出租车去参加城里的派对。 \
             在派对上，伊丽莎白晕倒并且被火速送往医院。']
# a list of human-annotated reference summaries
ref_list = ['伊丽莎白在与彼得参加完一个聚会后被送进医院。']
# a list of model outputs to be evaluataed
output_list = ['彼得和伊丽莎白参加聚会。伊丽莎白赶到医院。']
```

Output:
```
+-------------+----------+
|  Dimensions |  Score   |
+-------------+----------+
|  coherence  | 0.894565 |
| consistency | 0.923428 |
|   fluency   | 0.880475 |
|  relevance  | 0.942078 |
|   overall   | 0.910136 |
+-------------+----------+
```

File: applications/Chat/evaluate/README.md (hunk 2 above)
Comment (chengeharrison) [reply applications/Chat/evaluate/README.md:102]: Ok.

File: applications/Chat/evaluate/README.md (hunk 1 above)
Comment (chengeharrison) [reply applications/Chat/evaluate/README.md:21]: Ok.

**Review by TongLi3701 — APPROVED** [review]
Thanks Yuanchen.

**Pull request merged**